lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -393,17 +395,17 @@ def _requires_modification(self):
 
         # Check cache security groups
         cache_security_groups = []
-        for sg in self.data['CacheSecurityGroups']:
-            cache_security_groups.append(sg['CacheSecurityGroupName'])
+        for security_group in self.data['CacheSecurityGroups']:
+            cache_security_groups.append(security_group['CacheSecurityGroupName'])
         if set(cache_security_groups) != set(self.cache_security_groups):
             return True
 
         # check vpc security groups
-        if len(self.security_group_ids) > 0:
+        if self.security_group_ids:
             vpc_security_groups = []
             security_groups = self.data['SecurityGroups'] or []
-            for sg in security_groups:
-                vpc_security_groups.append(sg['SecurityGroupId'])
+            for security_group in security_groups:
```

Comment:
my linter did not like short variable names, so I changed it ;) can of course stay the same.